kernel/src/log_segment/tests.rs

```diff
@@ -374,9 +379,14 @@ fn build_snapshot_with_out_of_date_last_checkpoint() {
         Some(&checkpoint_metadata),
     );
 
-    let log_segment =
-        LogSegment::for_snapshot_impl(storage.as_ref(), log_root, Some(checkpoint_metadata), None)
-            .unwrap();
+    let log_segment = LogSegment::for_snapshot_impl(
+        storage.as_ref(),
+        log_root,
+        vec![],
```

Comment:
nit: comment what this is:
```suggestion
        vec![], // log-tail
```

same a bunch below